extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -38,28 +40,30 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		this._extensionContext = extensionContext;
 		this.books = [];
 		this.initialize(workspaceFolders.map(a => a.uri.fsPath));
-		vscode.commands.executeCommand('setContext', 'untitledBooks', openAsUntitled);
 		this.viewId = view;
 		this.prompter = new CodeAdapter();
 
 	}
 
 	private async initialize(bookPaths: string[]): Promise<any> {
+		await vscode.commands.executeCommand('setContext', 'untitledBooks', this._openAsUntitled);
 		return Promise.all(bookPaths.map(async (bookPath) => {
```

Comment:
I wouldn't even bother returning anything here and just have this be a Promise<void> since we're never actually using that here. Then you can just await the Promise.all and resolve the deferred right after.